subsys/net/lib/sockets/sockets.c

```diff
@@ -425,6 +425,7 @@ int zsock_accept_ctx(struct net_context *parent, struct sockaddr *addr,
 
```

Comment:
Please change the commit subject to `net: socket: Release fd on failure`
Not sure if it is needed but at least in my commits I have `Signed-off-by: ...` (yours was missing ":")
Also add `Fixes #22366` to commit body.
Please remove the sockets.c.bak from this PR.